store/sql_store.go

```diff
@@ -325,7 +109,39 @@ func (ss *SQLStore) Spinmint() SpinmintStore {
 func (ss *SQLStore) DropAllTables() {
 	err := ss.master.TruncateTables()
 	if err != nil {
-		mlog.Error("failed to drop all tabels", mlog.Err(err))
+		mlog.Error("failed to drop all tables", mlog.Err(err))
+	}
+}
+
+func runMigrations(db *sql.DB) {
+	// Create database driver
+	dbDriver, err := mysql.WithInstance(db, &mysql.Config{})
+	if err != nil {
+		mlog.Critical("Failed to create migration driver", mlog.Err(err))
+		os.Exit(1)
+	}
+	// Create source driver
+	s := bindata.Resource(migrations.AssetNames(), migrations.Asset)
+
+	srcDriver, err := bindata.WithInstance(s)
+	if err != nil {
+		mlog.Critical("Failed to create source instance", mlog.Err(err))
+		os.Exit(1)
+	}
+
+	m, err := migrate.NewWithInstance("go-bindata", srcDriver, "mysql", dbDriver)
+	if err != nil {
+		mlog.Critical("Failed to create db instance", mlog.Err(err))
+		os.Exit(1)
+	}
+	// Run migration
+	err = m.Up()
+	// We ignore if there is no change and if file does not exist.
+	// The latter occurs if we have rolled back to older code without running down migrations.
+	// This is not ideal, but not critical enough to bail execution.
+	if err != nil && err != migrate.ErrNoChange && err != os.ErrNotExist {
+		mlog.Critical("Failed to migrate DB", mlog.Err(err))
+		os.Exit(1)
```

Comment:
I can't believe I am saying this but wouldn't it be more *idiomatic* to use `errors.Is()`?

Reply:
Haha .. yeah for `ErrNotExist` definitely. :+1: May not be necessary for `ErrNoChange` since it's directly returned from the library itself and is publicly documented.